src/flb_pack.c

```diff
@@ -293,3 +293,193 @@ void flb_pack_print(char *data, size_t bytes)
     }
     msgpack_unpacked_destroy(&result);
 }
+
+
+static inline int try_to_write(char *buf, int *off, size_t left,
+                        char *str, size_t str_len)
+{
+    if (str_len <= 0){
+        str_len = strlen(str);
+    }
+    if (left <= *off+str_len) {
+        return FLB_FALSE;
+    }
+    memcpy(buf+*off, str, str_len);
+    *off += str_len;
+    return FLB_TRUE;
+}
+
+static int msgpack2json(char *buf, int *off, size_t left, msgpack_object *o)
+{
+    int ret = FLB_FALSE;
+    int i;
+    int loop;
+
+    switch(o->type) {
+    case MSGPACK_OBJECT_NIL:
+        ret = try_to_write(buf, off, left, "null", 4);
+        break;
+
+    case MSGPACK_OBJECT_BOOLEAN:
+        ret = try_to_write(buf, off, left,
+                           (o->via.boolean ? "true":"false"),0);
+        
+        break;
+
+    case MSGPACK_OBJECT_POSITIVE_INTEGER:
+        {
+            char temp[32] = {0};
```

Comment:
> allocating

Sorry, It is not proper word.
In your case, char temp is initialized in every cases.

https://gist.github.com/nokute78/993f8c647dcf6bef8e9feeda66af12c4#file-init_char-c
I wrote sample code, In `init_everytime`  "char buf" is function scope.

https://gist.github.com/nokute78/993f8c647dcf6bef8e9feeda66af12c4#file-objdump_init_everytime-txt
It is assembly code of `init_everytime` with gcc-arm.
84f8-8558 are instructions to initialize "char buf".
These instructions are executed every cases.

On the other hand, this is a assembly code of `init_at_case1`.
https://gist.github.com/nokute78/993f8c647dcf6bef8e9feeda66af12c4#file-objdump_init_at_case1-txt
8434- are instructions to initialize "char buf".

```
    841c:   e3530001    cmp r3, #1
    8420:   0a000003    beq 8434 <init_at_case1+0x34>
```
This means 8434 is a head of `case 1:`
So, initializing instructions are processed only `case 1:`